File: native/android/app/src/main/java/com/example/hybrid/bridge/plugins/AppPlugin.kt
package com.example.hybrid.bridge.plugins

import android.content.Context
import com.example.hybrid.bridge.BridgePlugin
import com.example.hybrid.bridge.BridgeResult
import org.json.JSONObject

class AppPlugin(private val context: Context) : BridgePlugin {
    override val namespace: String = "device"

    override fun handle(method: String, params: JSONObject, completion: (BridgeResult) -> Unit) {
        when (method) {
            "getPlatform" -> {
                completion(BridgeResult(0, "ok", JSONObject().apply {
                    put("platform", "android")
                    put("version", android.os.Build.VERSION.RELEASE)
                    put("appVersion", "1.0.0")
                    put("statusBarHeight", 24)
                    put("safeAreaInset", JSONObject().apply {
                        put("top", 24); put("bottom", 0); put("left", 0); put("right", 0)
                    })
                    put("apiBaseUrl", "https://api.example.com")
                    put("environment", "release")
                }))
            }
            "getDeviceId" -> {
                completion(BridgeResult(0, "ok", JSONObject().put("deviceId", android.os.Build.SERIAL ?: "unknown")))
            }
            "checkUpdate" -> {
                completion(BridgeResult(0, "ok", JSONObject().put("hasUpdate", false)))
            }
            else -> completion(BridgeResult(-1, "method not found", null))
        }
    }
}

File: native/ios/HybridApp/Bridge/Plugins/AppPlugin.swift
import Foundation

/// App 级能力插件：检查更新、平台信息
final class AppPlugin: BridgePlugin {
    var bridge: JSBridgeContentController?
    let namespace = "device"

    func handle(method: String, params: Any, completion: @escaping (BridgeResult) -> Void) {
        switch method {
        case "getPlatform":
            let info: [String: Any] = [
                "platform": "ios",
                "version": UIDevice.current.systemVersion,
                "appVersion": Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? "1.0.0",
                "statusBarHeight": UIApplication.shared.statusBarFrame.height,
                "safeAreaInset": [
                    "top": UIApplication.shared.windows.first?.safeAreaInsets.top ?? 0,
                    "bottom": UIApplication.shared.windows.first?.safeAreaInsets.bottom ?? 0,
                    "left": 0,
                    "right": 0
                ],
                "apiBaseUrl": "https://api.example.com",
                "environment": "release"
            ]
            completion(BridgeResult(code: 0, msg: "ok", data: info))
        case "getDeviceId":
            let id = UIDevice.current.identifierForVendor?.uuidString ?? "unknown"
            completion(BridgeResult(code: 0, msg: "ok", data: ["deviceId": id]))
        default:
            // app.checkUpdate
            if method == "checkUpdate" {
                completion(BridgeResult(code: 0, msg: "ok", data: ["hasUpdate": false]))
            } else {
                completion(BridgeResult(code: -1, msg: "method not found", data: nil))
            }
        }
    }
}
